### PR TITLE
MET-6103 Cleanup deprecations on http harvesting

### PR DESCRIPTION
**Pull request by stzanakis**

### CODE REVIEW

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java [hunk 1]
```diff
@@ -216,153 +193,5 @@ public void forEachFiltered(ReportingIteration<Path> action, Predicate<Path> fil
     public String getExtractedDirectory() {
       return super.getExtractedDirectory();
     }
-
-    @Override
-    public void deleteIteratorContent() {
```
Comment (jeortizquan) [inline metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java:221]: I see this is fulfilled by the `AutoClosable` used by `AbstractHttpHarvestIterator`

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java [hunk 2]
```diff
@@ -216,153 +193,5 @@ public void forEachFiltered(ReportingIteration<Path> action, Predicate<Path> fil
...
-      this.close();
-    }
-  }
-
-  /**
-   * Iterator for harvesting
-   */
-  private abstract static class AbstractHttpHarvestIterator<R> implements HarvestingIterator<R, Path> {
```
Comment (jeortizquan) [inline metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvesterImpl.java:229]: I see this is now an extracted class.

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/AbstractHttpHarvestIterator.java [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+    public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+      final Path dirName = dir.getFileName();
+      if (dirName != null && MAC_TEMP_FOLDER.equals(dirName.toString())) {
+        return FileVisitResult.SKIP_SUBTREE;
+      }
+      return FileVisitResult.CONTINUE;
+    }
+  }
+
+  record FullRecordImpl(String relativeFilePath, ByteArrayInputStream entryContent) implements FullRecord {
```
Comment (jeortizquan) [inline metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/AbstractHttpHarvestIterator.java:149]: should be this also `private` as before the class extraction?

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvester.java [hunk 4]
```diff
@@ -79,29 +50,7 @@ HttpRecordIterator createTemporaryHttpHarvestIterator(InputStream input,
...
    */
-  FullRecordHarvestingIterator<ArchiveEntry, Path> createFullRecordHarvestIterator(InputStream input,
+  FullRecordHarvestingIterator<FullRecord, Path> createFullRecordHarvestIterator(InputStream input,
       CompressedFileExtension compressedFileType) throws HarvesterException;
 
-  /**
-   * An object representing an entry in a file archive. The harvesting identifier is the file name
-   * (including the path relative to the archive root).
-   */
-  interface ArchiveEntry extends FullRecord {
```
Comment (jeortizquan) [inline metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvester.java:89]: this will impact the sandbox.

**Review by jeortizquan — COMMENTED** [review]
I have some question and observations.

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/AbstractHttpHarvestIterator.java [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+/**
+ * Iterator for harvesting
+ */
+abstract class AbstractHttpHarvestIterator<R> implements HarvestingIterator<R, Path> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+  private final Path extractedDirectory;
+
+  protected AbstractHttpHarvestIterator(Path extractedDirectory) {
+    if (extractedDirectory == null) {
```
[the remark this answers is not in the document]
Comment (stzanakis) [reply metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/AbstractHttpHarvestIterator.java:36]: Ok I'm adding it. Keep in mind I am resolving issues from Sonar, not necessarily modifying code.

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/AbstractHttpHarvestIterator.java (hunk 3 above)
Comment (stzanakis) [reply metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/AbstractHttpHarvestIterator.java:149]: yes I think it's better indeed

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvester.java (hunk 4 above)
Comment (stzanakis) [reply metis-harvesting/src/main/java/eu/europeana/metis/harvesting/http/HttpHarvester.java:89]: You are right! I will create a PR there as well then.

**Review by jeortizquan — APPROVED** [review]

**Pull request merged**